File: code/java/stack-queue/src/main/java/org/swufe/datastructure/Benchmark.java
package org.swufe.datastructure;

import java.util.Arrays;
import java.util.List;
import java.util.concurrent.ThreadLocalRandom;

public class Benchmark {
    private static long timeCQ(int n) {
        CircularArrayQueue<Integer> q = new CircularArrayQueue<>();
        for (int i = 0; i < n; i++) {
            q.enqueue(ThreadLocalRandom.current().nextInt(n));
        }
        long start = System.currentTimeMillis();
        // to avoid the randomness of single operation
        // we repeat it 20 times
        for (int i = 0; i < 20; i++) {
            q.dequeue();
        }
        long end = System.currentTimeMillis();
        return end - start;
    }

    public static void timeDequeue() {
        List<Integer> sizes = Arrays.asList(10000000, 20000000, 30000000, 40000000, 50000000);
        for (int size : sizes) {
            long t = timeCQ(size);
            System.out.println(size + ": " + t);
        }
    }

    public static void main(String[] args) {
        timeDequeue();
    }
}

File: code/python/stack-queue/benchmark.py
import random
import datetime
from queue import Queue


def queue_vary_size():
    def time_dequeue(n):
        q = Queue()
        for _ in range(n):
            q.enqueue(random.randint(1, n))
        start = datetime.datetime.now()
        # to avoid the randomness of single operation
        # we repeat it 20 times
        for _ in range(20):
            q.dequeue()
        end = datetime.datetime.now()
        return int(round((end - start).total_seconds() * 1000))

    with open('benchmark_queue.txt', 'w') as f:
        for size in [1000000, 2000000, 3000000, 4000000, 5000000]:
            t = time_dequeue(size)
            f.write(f'{size}   {t}\n')


queue_vary_size()
